### PR TITLE
[devel_transport] Async test cases porting

### PR DESCRIPTION
**Pull request by PrzemekWirkus**
# Description
First phase, porting or basic and generic test cases + RTC and timers.

This will be tested internally first with custom ```utest```.

### CODE REVIEW

File: test/c_strings/main.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+#define POSITIVE_INTEGERS 32768,3214,999,100,1,0,1,4231,999,4123,32760,99999
+#define FLOATS  0.002,0.92430,15.91320,791.77368,6208.2,25719.4952,426815.982588,6429271.046,42468024.93,212006462.910
+
+using namespace utest::v1;
+
+
+void test_case_c_string_i_d() {
+    CLEAN_BUFFER;
+    sprintf(buffer, "%i %d %i %d %i %d %i %d %i %d %i %i", NEGATIVE_INTEGERS);
+    TEST_ASSERT_FALSE(strcmp(buffer, "-32768 -3214 -999 -100 -1 0 -1 -4231 -999 -4123 -32760 -99999"));
```
Comment (niklas-arm) [inline test/c_strings/main.cpp]: Y u no `TEST_ASSERT_EQUAL_STRING_LEN`?

File: test/c_strings/main.cpp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+status_t greentea_failure_handler(const Case *const source, const failure_t reason) {
+    greentea_case_failure_abort_handler(source, reason);
+    return STATUS_CONTINUE;
+}
+
+Case cases[] = {
+    Case("C stings strtok", test_case_c_string_strtok, greentea_failure_handler),
+    Case("C stings strpbrk", test_case_c_string_strpbrk, greentea_failure_handler),
```
Comment (niklas-arm) [inline test/c_strings/main.cpp]: `stings`?

File: test/wait_us/main.cpp [hunk 3]
```diff
@@ -15,34 +15,37 @@
...
+Case cases[] = {
+    Case("Timers: wait_us", test_case_ticker),
+};
+
+status_t greentea_test_setup(const size_t number_of_cases) {
+    GREENTEA_SETUP(20, "wait_us_auto");
+    return greentea_test_setup_handler(number_of_cases);
 }
 
+Specification specification(greentea_test_setup, cases, greentea_test_teardown_handler);
```
Comment (niklas-arm) [inline test/wait_us/main.cpp]: Isn't `greentea_test_teardown_handler` the default behavior anyways?

**Comment by PrzemekWirkus** [comment]
@niklas-arm Thanks mate!

File: module.json [hunk 4]
```diff
@@ -35,7 +35,12 @@
     "ualloc": "^1.0.0",
     "minar": "^1.0.0",
     "core-util": "^1.0.0",
-    "compiler-polyfill": "^1.1.0"
+    "compiler-polyfill": "^1.1.0",
+    "greentea-client": "^0.1.4"
+  },
+  "testDependencies": {
+    "unity": "^2.0.0",
+    "greentea-client": "^0.1.4"
```
Comment (0xc0170) [inline module.json]: why gt client is dependency and also test dep?

Comment (PrzemekWirkus) [inline module.json]: For code coverage functions in retarget: [here](https://github.com/ARMmbed/mbed-drivers/blob/devel_transport/source/retarget.cpp#L188) and [here](https://github.com/ARMmbed/mbed-drivers/blob/devel_transport/source/retarget.cpp#L237).

**Pull request merged**